src/handlers/v1/markets/[chainId]/[marketId]/collateral-swap/index.ts

```diff
@@ -0,0 +1,150 @@
+        marketInfo;
+
+      // Verify token input
+      const _withdrawalToken = common.Token.from(withdrawalToken);
+      const withdrawalCollateral = collaterals.find(({ asset }) => asset.is(_withdrawalToken.unwrapped));
+      if (!withdrawalCollateral) {
+        throw newHttpError(400, { code: '400.5', message: 'withdrawal token is not collateral' });
+      }
+      const _targetToken = common.Token.from(targetToken);
+      const targetCollateral = collaterals.find(({ asset }) => asset.is(_targetToken.unwrapped));
+      if (!targetCollateral) {
+        throw newHttpError(400, { code: '400.5', message: 'target token is not collateral' });
+      }
+
+      // 1. new balancer flash loan logics and append loan logic
+      const [flashLoanLoanLogic, flashLoanRepayLogic] = apisdk.protocols.balancerv2.newFlashLoanLogicPair([
+        { token: _withdrawalToken, amount: amount },
+      ]);
+      logics.push(flashLoanLoanLogic);
+
+      // 2. new and append paraswap swap token logic
+      const quotation = await apisdk.protocols.paraswapv5.getSwapTokenQuotation(chainId, {
+        input: { token: _withdrawalToken, amount: amount },
+        tokenOut: _targetToken,
+        slippage,
+      });
+      logics.push(apisdk.protocols.paraswapv5.newSwapTokenLogic(quotation));
+
+      // 3. new and append compound v3 supply collateral logic
+      logics.push(
+        apisdk.protocols.compoundv3.newSupplyCollateralLogic({
+          marketId,
+          input: { token: _targetToken, amount: quotation.output.amount },
+          balanceBps: common.BPS_BASE,
+        })
+      );
+
+      // 4. new and append compound v3 withdraw logic
+      logics.push(
+        apisdk.protocols.compoundv3.newWithdrawCollateralLogic({
+          marketId,
+          output: { token: _withdrawalToken, amount: amount },
+        })
+      );
+
+      // 5. append balancer flash loan replay logic
+      logics.push(flashLoanRepayLogic);
+
+      const estimateResult = await apisdk.estimateRouterData({ chainId, account, logics });
+      approvals = estimateResult.approvals;
+
+      // 6. calc target position
+      const _withdrawalUSD = new BigNumberJS(amount).times(withdrawalCollateral.assetPrice);
+      const targetUSD = new BigNumberJS(quotation.output.amount).times(targetCollateral.assetPrice);
+      const targetBorrowUSD = new BigNumberJS(borrowUSD);
+      const targetCollateralUSD = new BigNumberJS(collateralUSD).minus(_withdrawalUSD).plus(targetUSD);
+      const targetBorrowCapacityUSD = new BigNumberJS(borrowCapacityUSD);
+
+      const targetLiquidationLimit = new BigNumberJS(liquidationLimit).minus(
+        _withdrawalUSD
+          .times(withdrawalCollateral.liquidateCollateralFactor)
+          .plus(targetUSD.times(targetCollateral.liquidateCollateralFactor))
+      );
+      const targetLiquidationThreshold = common.formatBigUnit(targetLiquidationLimit.div(targetCollateralUSD), 4);
+      targetPosition = {
```

Comment:
According to the PRD P.11, there should also be information about `isCollateral` and `liquidationThreshold`, and there is no `totalDebt`.